modules/apps/change-tracking/change-tracking-web/src/main/java/com/liferay/change/tracking/web/internal/portlet/action/UpdateCTCommentMVCResourceCommand.java

```diff
@@ -37,23 +42,46 @@ protected void doServeResource(
 			ResourceRequest resourceRequest, ResourceResponse resourceResponse)
 		throws Exception {
 
+		ThemeDisplay themeDisplay = (ThemeDisplay)resourceRequest.getAttribute(
+			WebKeys.THEME_DISPLAY);
+
+		long ctCollectionId = ParamUtil.getLong(
+			resourceRequest, "ctCollectionId");
+
+		if (!CTCollectionPermission.contains(
+				themeDisplay.getPermissionChecker(), ctCollectionId,
+				ActionKeys.VIEW)) {
+
+			JSONPortletResponseUtil.writeJSON(
+				resourceRequest, resourceResponse,
+				JSONUtil.put(
+					"errorMessage",
+					_language.get(
+						themeDisplay.getLocale(),
+						"you-do-not-have-the-required-permissions")));
+
+			return;
+		}
+
 		CTComment ctComment = null;
 
 		long ctCommentId = ParamUtil.getLong(resourceRequest, "ctCommentId");
 
 		String value = ParamUtil.getString(resourceRequest, "value");
 
 		if (ctCommentId > 0) {
-			ctComment = ctCommentLocalService.updateCTComment(
-				ctCommentId, value);
+			CTComment existingCTComment = ctCommentLocalService.getCTComment(
+				ctCommentId);
+
+			if (existingCTComment.getUserId() == themeDisplay.getUserId()) {
+				ctComment = ctCommentLocalService.updateCTComment(
+					ctCommentId, value);
+			}
+			else {
+				return;
```

Comment:
We should also write an error message here if the user is trying to modify other user's comment.